src/Sylius/Bundle/ApiBundle/CommandHandler/Checkout/AddressOrderHandler.php

```diff
@@ -72,7 +78,12 @@ public function __invoke(AddressOrder $addressOrder): OrderInterface
         );
 
         if (null === $order->getCustomer()) {
-            $order->setCustomer($this->provideCustomerByEmail($addressOrder->email));
+            /** @var CustomerInterface $customer */
+            $customer = $this->provideCustomerByEmail($addressOrder->email);
+
+            $order->setCustomer($customer);
+
+            $this->cartVisitorsCustomerContext->setCartCustomerId((string) $customer->getId());
```

Comment:
This is the follow-up comment to the question, why do we have `cartVisitorsCustomerContext` here in the first place. This is the only place it is called with meaningful logic. It is nulled in second place(PickupCarthandler). Why can we base our logic on `$order->setCustomer($customer);` and appropriate order state?